java-spiffe-core/src/main/java/spiffe/bundle/BundleSource.java

```diff
@@ -0,0 +1,22 @@
+package spiffe.bundle;
+
+
+import lombok.NonNull;
+import spiffe.bundle.x509bundle.X509Bundle;
+import spiffe.exception.BundleNotFoundException;
+import spiffe.spiffeid.TrustDomain;
+
+/**
+ * A <code>BundleSource</code> represents a source of bundles of type T keyed by trust domain.
+ */
+public interface BundleSource<T> {
+
+    /**
+     * Returns the bundle of type T associated to the given trust domain.
+     *
+     * @param trustDomain an instance of a {@link TrustDomain}
+     * @return the {@link X509Bundle} for the given trust domain
```

Comment:
This `@link` looks incorrect, it's pointing to a specific type, `X509Bundle`, when the return type is the generic type `T`.

Reply:
Fixed.